pexec/managed_process.go

```diff
@@ -358,7 +373,6 @@ func (p *managedProcess) Stop() error {
 		p.mu.Unlock()
 		return nil
 	}
-	p.mu.Unlock()
```

Comment:
Why has our section of `Lock`ed behavior expanded? And, if it's valid to expand it, I think you should just `defer p.mu.Unlock()` here instead of adding an `Unlock` to every return path.

Reply:
I tried defer initially. To prevent a race condition flagged by CI, but it still didn't pass. I did this just to be explicit and see if there was a difference and now it passes with this.

Reply:
Need to do further testing and see how I can pass CI with minimal changes. @maximpertsov is also tracking